### PR TITLE
Pulled in code from curate to manage add/remove fields on edit forms.  A...

### PR DESCRIPTION
**Pull request by val99erie**
I wasn't able to finish this story by the end of the sprint, so @cam156 asked me to open a pull request so that another developer can pick up the rest of the story.

As far as I know, there are 4 main things left to do:

.1. Fix disabled auto-complete

I replaced the javascript for creating the add/remove buttons next to each field in the edit form with the javascript that I took from curate.

The old javascript:  app/assets/javascripts/sufia/multiForm.js
The new javascript:  app/assets/javascripts/sufia/manage_repeating_fields.js

Sufia had some auto-completion behavior attached to those old fields, so I had to disable the auto-complete functionality while I was working on the views.  I left a TODO message in the file where it is disabled to make sure to go back and fix it:
app/assets/javascripts/sufia/edit_metadata.js

Once the autocomplete is fixed, you can probably delete app/assets/javascripts/sufia/multiForm.js entirely.

.2. Style changes

When you are editing a file, there are 3 tabs.  I have already updated the javascript, views, and styles for the "Descriptions" tab, but I haven't had time to work on the "Permissions" tab yet.  It looks like the "Permissions" tab has its own javascript.  That javascript seems to be working just fine, so maybe it's best not to mess with it, but you might want to update the styles of the adder buttons so that they look like the buttons on the "Descriptions" tab.

.3. Upgrade version of hydra-editor in gemspec

I ran ScholarSphere with sufia pointing to this branch.  Then I tried to edit a file in ScholarSphere, but I got an error about calling a method with the wrong number of arguments.

The error is caused by the partial_exists? method in hydra-editor
(app/helpers/concerns/records_helper_behavior.rb)

That method has 1 argument on master of hydra-editor, but it has 2 arguments in hydra-editor 0.4.0.

So, sufia's gemspec needs to be pegged to a higher version of hydra-editor, or else that method needs to be overridden in sufia.

.4. Add something to the release notes

People who upgrade sufia might experience breaking changes, especially if they have overridden the views.

### CODE REVIEW

**Comment by mjgiarlo** [comment]
How much of this was superseded by #819, @jcoyne ?  Should we keep it open?

**Comment by jcoyne** [comment]
@mjgiarlo all of this will be superceded by #819, but this targets Sufia 5 and #819 is for Sufia 6

**Comment by cam156** [comment]
This is ready for commenting. The auto complete has been fixed.  I think we will leave permissions alone for the moment, and I think it would be better to update ScholarSphere to the latest hydra-editor, instead of pinning sufia down.

File: app/inputs/multi_value_input.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  def input(wrapper_options)
+    @rendered_first_element = false
+    input_html_classes.unshift("string")
+    #input_html_options[:type] ||= 'text'
+    input_html_options[:name] ||= "#{object_name}[#{attribute_name}][]"
+
+    markup = <<-HTML
+
+
```
Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: Are these two blank lines significant?

File: app/inputs/multi_value_input.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+class MultiValueInput < SimpleForm::Inputs::CollectionInput
+
+  def input(wrapper_options)
+    @rendered_first_element = false
+    input_html_classes.unshift("string")
+    #input_html_options[:type] ||= 'text'
```
Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: Remove?

File: app/inputs/multi_value_input.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    input_dom_id + '_label'
+  end
+
+  def input_dom_id
+    input_html_options[:id] || "#{object_name}_#{attribute_name}"
+  end
+
+  def collection
+    @collection ||= begin
+      Array(object.send(attribute_name))
```
Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: Why is this line in a begin...end block?

File: app/inputs/multi_value_input.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          </li>
+        </ul>
+
+    HTML
+  end
+
+
+  private
+
+  def build_field(value)
```
Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: This method is a bit beefy -- I might like to see it broken down a bit, perhaps into smaller methods, each of which does a single thing.

File: app/inputs/multi_value_input.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    input_html_options[:id] || "#{object_name}_#{attribute_name}"
+  end
+
+  def collection
+    @collection ||= begin
+      Array(object.send(attribute_name))
+    end
+  end
+
+  def multiple?; true; end
```
Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: Break into three lines?

File: app/views/records/edit_fields/_type.html.erb [hunk 6]
```diff
@@ -1,8 +1,3 @@
...
-        <%= f.label :resource_type, "Resource Types", for: "resource_type" %>
-        <p class="help-block">You may select multiple types to apply<p>
-        <%= f.select :resource_type, options_for_select(Sufia.config.resource_types.keys), {},
-          {multiple: true, size: 7} %>
-        <%= help_icon(:resource_type) %>
-      </div>
-
+<% required = render_req && record.required?(key) %>
+<%= render partial: "records/edit_fields/resource_type", locals: { record: record, f: f, required: required, key: key }
+%>
```
Comment (mjgiarlo) [inline app/views/records/edit_fields/_type.html.erb]: move to end of prior line?

File: app/inputs/multi_value_input.rb (hunk 3 above)
Comment (cam156) [inline app/inputs/multi_value_input.rb]: Not really sure.  I can try removing it.

File: app/inputs/multi_value_input.rb (hunk 4 above)
Comment (cam156) [inline app/inputs/multi_value_input.rb]: @mjgiarlo Split out the options building.  Does that satisfy?

Comment (mjgiarlo) [inline app/inputs/multi_value_input.rb]: :clap: Thanks!

**Comment by mjgiarlo** [comment]
:+1:

**Comment by mjgiarlo** [comment]
@jcoyne :grey_question:

**Comment by awead** [comment]
:clap: tutti bravi!

**Pull request merged**